### PR TITLE
feat: remove Mariner V1 target and pin SIG version

### PR DESCRIPTION
**Pull request by hbeberman**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Mariner 1.0 has reached end of life and is no longer receiving security updates or support; therefore, we should not be building newer VHDs for it anymore.
By extension the Mariner V1 AKS container host is end of life as well, those using it should move to using the AzureLinux OSSKU to continue to receive security updates and support.

**Which issue(s) this PR fixes**:

Mariner 1.0 is end of life and is no longer receiving updates or support.

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
Stop Mariner V1 (OSSKU cblmariner) image builds, new deployments should use the AzureLinux OSSKU instead.
```

### CODE REVIEW

File: e2e/scenario/init.go [hunk 1]
```diff
@@ -23,20 +23,16 @@ func scenarios() []*Scenario {
 	return []*Scenario{
 		ubuntu1804(),
 		ubuntu2204(),
-		marinerv1(),
```
Comment (cameronmeissner) [inline e2e/scenario/init.go:26]: this'll be good to free up scenario space for AzureLinux/cgroupsv2 e2e's

File: pkg/agent/datamodel/sig_config.go [hunk 2]
```diff
@@ -480,7 +484,7 @@ var (
 		ResourceGroup: AKSCBLMarinerResourceGroup,
 		Gallery:       AKSCBLMarinerGalleryName,
 		Definition:    "V1",
-		Version:       LinuxSIGImageVersion,
+		Version:       FrozenCBLMarinerV1SIGImageVersionForDeprecation,
```
Comment (cameronmeissner) [inline pkg/agent/datamodel/sig_config.go:487]: oh, I guess we never published v1gen2...

**Review by cameronmeissner — APPROVED** [review]

File: pkg/agent/datamodel/sig_config.go (hunk 2 above)
Comment (anujmaheshwari1) [reply pkg/agent/datamodel/sig_config.go:487]: yeah we didnt but we always built it lol

**Review by anujmaheshwari1 — APPROVED** [review]
lgtm, thanks for the cleanup Henry!

**Pull request merged**